src/lib.rs

```diff
@@ -512,13 +512,43 @@ where
         self.borrow_mut()
     }
 
+    /// Resizes the `SecVec` in-place so that len is equal to `new_len`.
+    ///
+    /// If `new_len` is smaller the inner vector is truncated.
+    /// If `new_len` is larger the inner vector will grow, placing `value` in all new cells.
+    ///
+    /// This ensures that the new memory region is secured if reallocation occurs.
+    ///
+    /// Similar to [`Vec::resize`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize)
+    pub fn resize(&mut self, new_len: usize, value: T) {
+        let old_ptr = self.content.as_mut_ptr();
+        let old_cap = self.content.capacity();
+
+        self.content.resize(new_len, value);
+
+        let new_ptr = self.content.as_mut_ptr();
+        let new_cap = self.content.capacity();
+
+        // Zero old memory region if vec pointer changed
+        if old_ptr != new_ptr {
+            unsafe {
+                zero_vec_ptr_len(old_ptr, old_cap);
+            }
+        }
+
+        // Relock changed memory region if vec pointer/capacity changed
```

Comment:
I'd just unlock before the resize, and relock after the resize whatever the current pointer is